ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/results/tracing/flat/FlatTraceGenerator.java

```diff
@@ -530,11 +532,19 @@ private static FlatTrace.Context handleCallDataLoad(
 
   private static boolean hasRevertInSubCall(
       final TransactionTrace transactionTrace, final TraceFrame callFrame) {
-    return transactionTrace.getTraceFrames().stream()
-        .filter(traceFrame -> !traceFrame.equals(callFrame))
-        .takeWhile(traceFrame -> !traceFrame.getOpcode().equals("RETURN"))
-        .filter(traceFrame -> traceFrame.getOpcode().equals("REVERT"))
-        .anyMatch(traceFrame -> traceFrame.getDepth() == callFrame.getDepth());
+    for (int i = 0; i < transactionTrace.getTraceFrames().size(); i++) {
+      if (i + 1 < transactionTrace.getTraceFrames().size()) {
```

Comment:
Kind of a random question (Side note: I dont really have Java exp) but what is the point of this If condition? Wouldn't starting the loop at 1 and running it until `size() - 1` have the same result? (Or if iterating in reverse has no side-effects, even starting at size() - 1 and running while its > 0)